### PR TITLE
tokens: fix deposit permissions

### PR DESCRIPTION
**Pull request by slint**

### CODE REVIEW

File: zenodo/modules/records/permissions.py [hunk 1]
```diff
@@ -206,18 +206,32 @@ class DepositFilesPermission(object):
...
-        return self.func(current_user, self.record)
+        return self.func(self.user, self.record)
 
     @classmethod
     def create(cls, record, action):
         """Record and instance."""
+        rat_token = request.args.get('token')
+        if rat_token and action in cls.rat_actions:
+            rat_signer, payload = decode_rat(rat_token)
+            rat_depost_id = payload.get('deposit_id')
```
Comment (Glignos) [inline zenodo/modules/records/permissions.py:230]: typo

File: zenodo/modules/records/permissions.py [hunk 2]
```diff
@@ -355,8 +369,10 @@ def has_read_files_permission(user, record):
     rat_token = request.args.get('token')
     if rat_token:
         rat_signer, payload = decode_rat(rat_token)
+        rat_depost_id = payload.get('deposit_id')
+        rat_access = payload.get('access')
         deposit_id = record.get('_deposit', {}).get('id')
-        if payload['deposit_id'] == deposit_id:
+        if rat_depost_id == deposit_id and rat_access == 'read':
```
Comment (Glignos) [inline zenodo/modules/records/permissions.py:375]: Change `rat_access` check to a more flexible implementation as discussed IRL.

**Review by Glignos — COMMENTED** [review]
Almost there amigo!
image

File: tests/unit/deposit/test_api_files.py [hunk 3]
```diff
@@ -239,3 +242,64 @@ def test_deposit_deletion(api_client, deposit, json_auth_headers, deposit_url,
...
     res = client.get(download_link)
     assert res.status_code == 404
+
+
+def test_rat_deposit_files_access(
+        app, db, api_client, deposit, deposit_file, deposit_url,
+        json_auth_headers, license_record, rat_generate_token):
+    """Test deposit files access via RATs."""
+    client = api_client
+    rat_generate_token = rat_generate_token['token']
```
[the remark this answers is not in the document]
Comment (slint) [reply tests/unit/deposit/test_api_files.py:252]: In the end, I changed the fixture to just return the "RAT generate" token only (i.e. no need for the auth headers)

**Pull request merged**